root/static/scripts/edit/URLCleanup.js

```diff
@@ -915,6 +915,40 @@ const CLEANUPS: CleanupEntries = {
       return url.replace(/^(https:\/\/archive\.org\/details\/[A-Za-z0-9._-]+)\/$/, '$1');
     },
   },
+  'audiomack': {
+    match: [new RegExp('^(https?://)?([^/]+\\.)?audiomack\\.com/', 'i')],
+    restrict: [LINK_TYPES.streamingfree],
+    clean: function (url) {
+      url = url.replace(/^https?:\/\/(www.)?audiomack.com\//, 'https://audiomack.com/');
+      return url;
+    },
+    validate: function (url, id) {
+      const m = /^https:\/\/audiomack\.com\/([a-zA-Z0-9_-]+)(?:\/(album|song)\/[a-zA-Z0-9_-]+\/?)?$/.exec(url);
+      if (m) {
+        const artist = m[1];
+        const prefix = m[2];
+        switch (id) {
+          case LINK_TYPES.streamingfree.artist:
+            return {
+              result: !!artist && !prefix,
+              target: ERROR_TARGETS.ENTITY,
+            };
+          case LINK_TYPES.streamingfree.release:
+            return {
+              result: !!artist && prefix === 'album',
+              target: ERROR_TARGETS.ENTITY,
+            };
+          case LINK_TYPES.streamingfree.recording:
+            return {
+              result: !!artist && prefix === 'song',
+              target: ERROR_TARGETS.ENTITY,
+            };
+        }
+        return {result: false, target: ERROR_TARGETS.ENTITY};
+      }
+      return {result: false, target: ERROR_TARGETS.URL};
+    },
```

Comment:
Capturing artist seems to be useless here.
```suggestion
    validate: function (url, id) {
      const m = /^https:\/\/audiomack\.com\/[a-zA-Z0-9_-]+(?:\/(album|song)\/[a-zA-Z0-9_-]+\/?)?$/.exec(url);
      if (m) {
        const prefix = m[1];
        switch (id) {
          case LINK_TYPES.streamingfree.artist:
            return {
              result: !prefix,
              target: ERROR_TARGETS.ENTITY,
            };
          case LINK_TYPES.streamingfree.release:
            return {
              result: prefix === 'album',
              target: ERROR_TARGETS.ENTITY,
            };
          case LINK_TYPES.streamingfree.recording:
            return {
              result: prefix === 'song',
              target: ERROR_TARGETS.ENTITY,
            };
        }
        return {result: false, target: ERROR_TARGETS.ENTITY};
      }
      return {result: false, target: ERROR_TARGETS.URL};
    },
```